terminus/tests/vertex_graph_to_roads_converter_test.py

```diff
@@ -213,14 +229,53 @@ def test_get_roads_matrix_distribution(self):
         self.converter = VertexGraphToRoadsConverter(
             0.25, [v1, v2, v3, v4, v5, v6, v7, v8, v9])
         roads = self.converter.get_roads()
-        expected_roads = [
-            Street.from_points([Point(0, 0), Point(0, 1), Point(0, 2)]),
-            Street.from_points([Point(1, 0), Point(1, 1), Point(1, 2)]),
-            Street.from_points([Point(2, 0), Point(2, 1), Point(2, 2)]),
-            Street.from_points([Point(0, 0), Point(1, 0), Point(2, 0)]),
-            Street.from_points([Point(0, 1), Point(1, 1), Point(2, 1)]),
-            Street.from_points([Point(0, 2), Point(1, 2), Point(2, 2)])
-        ]
+        expected_roads = []
+        street = Street.from_points([Point(0, 0), Point(0, 1), Point(0, 2)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(1, 0), Point(1, 1), Point(1, 2)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(2, 0), Point(2, 1), Point(2, 2)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(0, 0), Point(1, 0), Point(2, 0)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(0, 1), Point(1, 1), Point(2, 1)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(0, 2), Point(1, 2), Point(2, 2)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        self.assertItemsEqual(roads, expected_roads)
+
+    def test_get_roads_Y_street_and_trunk_type(self):
+        """
+        (0,0)--(1,0)--(6,1)
+                  |---(6,-0.8)(trunk)
+        Note that (6,-0.8) is discarded because it's a trunk even though it has
+        a has a smaller angle but we are building a street type.
+        """
+        v1 = Vertex(Point(0, 0))
+        v2 = Vertex(Point(1, 0))
+        v3 = Vertex(Point(6, 1))
+        v4 = Vertex(Point(6, -0.8))
+        v4.minor_road = False  # make trunk
+
+        self._connect(v1, v2)
+        self._connect(v2, v3)
+        self._connect(v2, v4)
+
+        self.converter = VertexGraphToRoadsConverter(0.25, [v1, v2, v3, v4])
+        roads = self.converter.get_roads()
+        expected_roads = []
+        street = Street.from_points([Point(0, 0), Point(1, 0), Point(6, 1)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
+        street = Street.from_points([Point(1, 0), Point(6, -0.8)])
+        street.set_width(STREET_WIDTH)
+        expected_roads.append(street)
```

Comment:
Regarding your question, the edge between (1, 0) and (6, -0.8) should be a street because we always build in the street-->trunk direction. So streets can end in trunks but trunks can't end in streets.
I'll add more tests as you suggest.